### PR TITLE
Llu/move reduction scheduler

### PR DESCRIPTION
**Pull request by liqiangxl**
This PR did step-4 described in #864. It did two things:
(1) move `ReductionScheduler` from `registry.cpp` to `reduction.cpp`
(2) removed unnecessary header file inclusions in `reduction.cpp`

### CODE REVIEW

**Comment by liqiangxl** [comment]
!build

File: csrc/scheduler/reduction.cpp [hunk 1]
```diff
@@ -5,23 +5,15 @@
...
 #include <scheduler/reduction_utils.h>
-#include <scheduler/registry.h>
+#include <scheduler/registry_utils.h>
 #include <scheduler/utils.h>
 #include <scheduler/vectorize_helper.h>
-#include <transform_replay.h>
-
-#include <ir/iostream.h>
-
-#include <ATen/cuda/CUDAContext.h>
```
Comment (liqiangxl) [inline csrc/scheduler/reduction.cpp:24]: Discard headers that aren't needed.

File: csrc/scheduler/reduction.cpp [hunk 2]
```diff
@@ -853,6 +845,179 @@ std::shared_ptr<ReductionParams> outerReductionHeuristic(
 
 } // namespace
 
+ReductionScheduler::ReductionScheduler(
```
Comment (liqiangxl) [inline csrc/scheduler/reduction.cpp:848]: moved from registry.cpp without change.

**Comment by liqiangxl** [comment]
!build

**Comment by liqiangxl** [comment]
!build

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**